### PR TITLE
[FLOSS T2] Add validation to run_all_tests_with_docker tutorial

### PR DESCRIPTION
**Pull request by Bujuhu**
Co-authored-by: Dynamichost96 <60710121+Dynamichost96@users.noreply.github.com>

Additionally, we've updated the tutorial a bit, and removed redundant podman steps and added a notice on the top, that podman can be used instead of docker. 

We've manually disabled validations within the CI pipeline, as it would be very long, resource intensive, and I'm not sure if we even have access to the docker command within the environment.

If anyone has an idea how we could sensibly automate the validation of this tutorial, please let me know.
For now, reviews, if the manual validation of this script succeeds (`./build/tests/shell/shell_recorder/tutorial_wrapper/markdown_shell_recorder.sh doc/tutorials/run_all_tests_with_docker.md`) are greatly appreciated!

We worked on this issue entirely in pair

## Basics

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation (see [Documentation Guidelines](https://www.libelektra.org/devgettingstarted/documentation))
- [ ] I fixed all affected decisions (see [Decision Process](https://www.libelektra.org/decisions/decision-process))
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://www.libelektra.org/devgettingstarted/coding))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

File: doc/tutorials/run_all_tests_with_docker.md [hunk 1]
```diff
@@ -167,14 +145,14 @@ After starting the container, you should be automatically inside it in the worki
...
 Create a folder where Elektra will be installed, create another folder for building the source and `cd` to it and like this:
 
-```sh
+```shell
 mkdir elektra-install && mkdir elektra-build-docker && cd elektra-build-docker
 ```
 
 Build it with
 
 ```sh
```
Comment (0x6178656c) [inline doc/tutorials/run_all_tests_with_docker.md:154]:
```suggestion
```shell
```

I guess this should not be part of the shell recorder tests.

**Review by 0x6178656c — CHANGES_REQUESTED** [review]
Excellent improvement to the docker tutorial.

I think it makes sense to remove the Podman commands when they are equivalent to the Docker commands.

I am not entirely convinced that it makes sense to add a shell recorder test for this tutorial. As you have pointed out, the build environment in CI (probably) does not have Docker/Podman installed and therefore the tests would always fail. I think the shell recorder is mainly intended for testing functionality of an installation of Elektra as opposed to the build/development process thereof.

My suggestion would be to drop the test and change all fences to either `sh` or `shell`. In any case your changes have improved the tutorial.

**Comment by Bujuhu** [comment]
Do others agree, that shell recorder tests make no sense in this case? Then I'll remove it. I think it has *some* value, as the test could still be validated locally.

**Comment by Eiskasten** [comment]
Good Job, everything seems fine to me!

Regarding Docker within the CI:
Maybe there is some tool with the exact same command line interface such as docker but only acts as a simulator?
The intention of such a tool would be to validate the docker cli usage and the syntax of dockerfiles without actually running or building something.
Do you know if something like that exists?
If so, it can be discussed if it is worth to install within the build containers.

**Review by Eiskasten — COMMENTED** [review]
Good Job, everything seems fine to me!

Regarding Docker within the CI:
Maybe there is some tool with the exact same command line interface such as docker but only acts as a simulator?
The intention of such a tool would be to validate the docker cli usage and the syntax of dockerfiles without actually running or building something.
Do you know if something like that exists?
If so, it can be discussed if it is worth to install within the build containers.

File: tests/shell/shell_recorder/tutorial_wrapper/CMakeLists.txt [hunk 2]
```diff
@@ -34,6 +34,9 @@ add_msr_test (tutorial_storage_plugins "${CMAKE_SOURCE_DIR}/doc/tutorials/storag
 add_msr_test (cmerge "${CMAKE_SOURCE_DIR}/doc/tutorials/cmerge.md" REQUIRED_PLUGINS hosts line PROPERTY LABELS memleak)
 add_msr_test (tutorial_namespaces "${CMAKE_SOURCE_DIR}/doc/tutorials/namespaces.md")
 
+# This is test is very long running and very resource intensive. Please run this run validation for this tutorial manually
```
Comment (deoknats861) [inline tests/shell/shell_recorder/tutorial_wrapper/CMakeLists.txt:37]: ### Review

I think there is a redundant 'is':
```suggestion
# This test is very long running and very resource intensive. Please run this run validation for this tutorial manually
```

**Documentation** 
The documentation is clear and understandable. I think the removal of the redundant docker/podman steps is the right approach for refactoring this tutorial.

**Examples**
The examples are well chosen and understandable.

**Code is conforming to our Coding Guidelines**
Yes.

**APIs are conforming to our Design Guidelines**
Yes.

**Code is consistent to our Design Decisions**
Yes.

**Comment by Bujuhu** [comment]
@deoknats861 Thank you! I've improved the comment text

**Comment by Bujuhu** [comment]
> Good Job, everything seems fine to me!
> 
> Regarding Docker within the CI: Maybe there is some tool with the exact same command line interface such as docker but only acts as a simulator? The intention of such a tool would be to validate the docker cli usage and the syntax of dockerfiles without actually running or building something. Do you know if something like that exists? If so, it can be discussed if it is worth to install within the build containers.

I haven't found a solution specifically for docker, and none that would automatically reflect that the tutorial would become invalid, should docker CLI arguments change. It would certainly be possible to parse `docker --help` pages for usage definitions, and then check, if the tutorial satisfies it.

**Comment by flo91** [comment]
jenkins build libelektra please

**Comment by flo91** [comment]
Good improvement, thank you!

**Pull request merged**